lib/oaisys/exceptions.rb

```diff
@@ -0,0 +1,20 @@
+module Oaisys
+  class PMHError < StandardError
```

Comment:
(File name should be errors.rb, probably)

Since these two classes are directly in the Oaisys namespace (eg. the classes are `Oaisys::PMHError` and `Oaisys::BadArgumentError`) they should either each be in their own file directly in the lib/oaisys folder, or they could be in this file (`lib/oaisys/errors.rb`) if you nest them in a namespace for the file:
```ruby
module Oaisys::Errors
  class PMHError < StandardError
# etc
```

and then change the raise and rescue to use Errors::BadArgumentError. Either way works.

Reply:
Done